src/libraries/System.Private.CoreLib/src/System/Number.NumberToFloatingPointBits.cs

```diff
@@ -113,6 +149,665 @@ public FloatingPointInfo(ushort denormalMantissaBits, ushort exponentBits, int m
             1e22,   // 10^22
         };
 
+        private static readonly ulong[] s_Pow5128Table = {
```

Comment:
A good candidate for the upcoming C# feature to store such arrays in the data section via ReadOnlySpan 🙂

Reply:
yes and in 2nd step of my PR, i'll adapt some other optimisations that are actually part of csFastFloat lib.

Reply:
in csFastFloat, we've used ReadOnly spans to hold bytes arrays, shall I use it for powers tables ?

Reply:
I think you can change it to 
```csharp
private static ReadOnlySpan<ulong> Pow5128Table => new ulong[] {
```
and wait Roslyn to optimize it (should happen soon, the runtime support was recently merged: https://github.com/dotnet/runtime/pull/61079).

So technically soon these two methods will have the same codegen: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBpuAE8MYACAFRlw3IAoBLAHYYWggCYwEAShYBvGiwUsASjGyiA8gIA2DAMoAHbAIA8jZgD4WGbMC2sAvCwEwA7izMwA2gF1ZLcmikaADMaOgArCwAvgDc8orEAOxWNnaeYhLecdRRdLTUAK5aEAIA5uycGKT8QiIC4lKy8Qoqapo6BkbGRSWllta2Dk6uLD1lPn4BQaER0dmKLEkpg+n1mdlRQA

Reply:
As we store power tables as ReadOnlySpans, is it a good idea to access elements with :
```
 ref byte tableRef = ref MemoryMarshal.GetReference(powersTable);
 return Unsafe.AddByteOffset(ref tableRef, (nint)n);
```
Adjusted to double and ulong ?

Reply:
Only if a normal index access emits a bound check (and it wasn't eliminated) that affects performance

Reply:
I don't believe this will be the case here as there's no table lookup inside a loop...  I'll leave it as is for now.